src/index.js

```diff
@@ -20,15 +20,99 @@ function createElement(tagName, props, ...children) {
   return element;
 }
 
-function render() {
+function render({
+  prevInput, savedOperator, savedValue, currentValue,
+}) {
+  function compute(prevValue, newValue) {
+    return {
+      '+': prevValue + newValue,
+      '-': prevValue - newValue,
+      '*': prevValue * newValue,
+      '/': prevValue / newValue,
+    };
+  }
+
+  function areNumbers(input1, input2) {
+    return typeof (input1) === 'number' && typeof (input2) === 'number';
+  }
+
+  function handleNumber(input) {
+    if (areNumbers(prevInput, input)) {
+      const value = currentValue * 10 + input;
+
+      render({
+        prevInput: input, savedOperator, savedValue, currentValue: value,
+      });
+      return;
+    }
+    render({
+      prevInput: input, savedOperator, savedValue: currentValue, currentValue: input,
+    });
+  }
+
+  function handleEquality(input) {
+    const value = compute(savedValue, currentValue)[savedOperator];
+
+    render({
+      prevInput: input, savedOperator: null, savedValue: null, currentValue: value,
+    });
+  }
+
+  function haveSavedOperator() {
+    return !!savedOperator;
+  }
+
+  function handleOperator(input) {
+    if (haveSavedOperator()) {
+      const value = compute(savedValue, currentValue)[savedOperator];
+
+      render({
+        prevInput: input, savedOperator: input, savedValue: null, currentValue: value,
+      });
+      return;
+    }
+    render({
+      prevInput: input, savedOperator: input, savedValue, currentValue,
+    });
+  }
+
+  function handleSign(input) {
+    if (input === '=') {
+      handleEquality(input);
+      return;
+    }
+    handleOperator(input);
+  }
+
+  const [zero, ...rest] = [...Array(10).keys()];
+
   const element = (
     <div>
       <p>간단 계산기</p>
+      <p>{currentValue}</p>
+      <p>
+        {[...rest, zero].map((i) => (
+          <button type="button" onClick={() => handleNumber(i)}>
+            {i}
+          </button>
+        ))}
+      </p>
+      <p>
+        {['+', '-', '*', '/', '='].map((sign) => (
+          <button type="button" onClick={() => handleSign(sign)}>
+            {sign}
+          </button>
+        ))}
+      </p>
     </div>
   );
 
-  document.getElementById('app').textContent = '';
-  document.getElementById('app').appendChild(element);
+  const app = document.getElementById('app');
+  app.textContent = '';
+  app.appendChild(element);
 }
 
-render();
+const defaultValue = {
+  prevInput: null, savedOperator: null, savedValue: null, currentValue: 0,
+};
```

Comment:
```suggestion
const defaultValue = {
  prevInput: null,
  savedOperator: null,
  savedValue: null,
  currentValue: 0,
};
```

초기값을 잘 분리해주셨어요 👍 
이러한 변수들은 상단에 주로 선언해주는 편이에요!